diff --git a/src/mem/mod.rs b/src/mem/mod.rs
index ccf1917..dcbbf52 100644
--- a/src/mem/mod.rs
+++ b/src/mem/mod.rs
@@ -130,6 +130,7 @@ fn memset_element_unordered_atomic<T>(s: *mut T, c: u8, bytes: usize)
     }
 }
 
+#[cfg(not(target_arch = "m68k"))]
 intrinsics! {
     #[cfg(target_has_atomic_load_store = "8")]
     pub unsafe extern "C" fn __llvm_memcpy_element_unordered_atomic_1(dest: *mut u8, src: *const u8, bytes: usize) -> () {
