diff --git a/src/thread/mod.rs b/src/thread/mod.rs
index e4581c2..4ff9d03 100644
--- a/src/thread/mod.rs
+++ b/src/thread/mod.rs
@@ -1099,7 +1099,7 @@ fn exhausted() -> ! {
         }

         cfg_if::cfg_if! {
-            if #[cfg(target_has_atomic = "64")] {
+            if #[cfg(all(target_has_atomic = "64", not(target_arch = "sparc")))] {
                 use crate::sync::atomic::{AtomicU64, Ordering::Relaxed};

                 static COUNTER: AtomicU64 = AtomicU64::new(0);
