diff --git a/src/sys/pal/unix/stack_overflow.rs b/src/sys/pal/unix/stack_overflow.rs
index 78a5990..1b875e5 100644
--- a/src/sys/pal/unix/stack_overflow.rs
+++ b/src/sys/pal/unix/stack_overflow.rs
@@ -213,6 +213,9 @@ pub unsafe fn drop_handler(data: *mut libc::c_void) {
     fn sigstack_size() -> usize {
         // FIXME: reuse const from libc when available?
         const AT_MINSIGSTKSZ: crate::ffi::c_ulong = 51;
+        #[cfg(target_env = "uclibc")]
+        let dynamic_sigstksz: crate::ffi::c_ulong = 0;
+        #[cfg(not(target_env = "uclibc"))]
         let dynamic_sigstksz = unsafe { libc::getauxval(AT_MINSIGSTKSZ) };
         // If getauxval couldn't find the entry, it returns 0,
         // so take the higher of the "constant" and auxval.
