diff --git a/debian/control.template b/debian/control.template
index 392ef08..d844772 100644
--- a/debian/control.template
+++ b/debian/control.template
@@ -1,7 +1,6 @@
 Package: @@PACKAGE@@-mingw-w64
 Architecture: all
 Depends: @@PACKAGE@@-mingw-w64-i686,
-         @@PACKAGE@@-mingw-w64-x86-64,
          ${misc:Depends}
 Recommends: @@RECOMMENDS@@
 Built-Using: gcc-@@VERSION@@ (= ${gcc:Version})
@@ -68,58 +67,3 @@ Description: GNU @@LANGUAGE@@ compiler for MinGW-w64, Win32/Win32
  cross-compiling to 32-bit MinGW-w64 targets, using the Win32
  threading model.
 Build-Profiles: <!stage1>
-
-Package: @@PACKAGE@@-mingw-w64-x86-64
-Architecture: all
-Depends: @@PACKAGE@@-mingw-w64-x86-64-posix,
-         @@PACKAGE@@-mingw-w64-x86-64-win32,
-         ${misc:Depends}
-Built-Using: gcc-@@VERSION@@ (= ${gcc:Version})
-Description: GNU @@LANGUAGE@@ compiler for MinGW-w64 targeting Win64
- MinGW-w64 provides a development and runtime environment for 32- and
- 64-bit (x86 and x64) Windows applications using the Windows API and
- the GNU Compiler Collection (gcc).
- .
- This metapackage provides the @@LANGUAGE@@ compiler, supporting
- cross-compiling to 64-bit MinGW-w64 targets.
-Build-Profiles: <!stage1>
-
-Package: @@PACKAGE@@-mingw-w64-x86-64-posix
-Architecture: @@ARCH@@
-Depends: @@DEPENDS64P@@,
-         ${misc:Depends},
-         ${shlibs:Depends}
-Suggests: gcc-@@VERSION@@-locales (>= ${local:Version})
-Breaks: @@BREAKS64@@
-Conflicts: @@CONFLICTS64@@
-Replaces: @@REPLACES64@@
-Built-Using: gcc-@@VERSION@@ (= ${gcc:Version})
-Description: GNU @@LANGUAGE@@ compiler for MinGW-w64, Win64/POSIX
- MinGW-w64 provides a development and runtime environment for 32- and
- 64-bit (x86 and x64) Windows applications using the Windows API and
- the GNU Compiler Collection (gcc).
- .
- This package contains the @@LANGUAGE@@ compiler, supporting
- cross-compiling to 64-bit MinGW-w64 targets, using the POSIX
- threading model.
-Build-Profiles: <!stage1>
-
-Package: @@PACKAGE@@-mingw-w64-x86-64-win32
-Architecture: @@ARCH@@
-Depends: @@DEPENDS64W@@,
-         ${misc:Depends},
-         ${shlibs:Depends}
-Suggests: gcc-@@VERSION@@-locales (>= ${local:Version})
-Breaks: @@BREAKS64@@
-Conflicts: @@CONFLICTS64@@
-Replaces: @@REPLACES64@@
-Built-Using: gcc-@@VERSION@@ (= ${gcc:Version})
-Description: GNU @@LANGUAGE@@ compiler for MinGW-w64, Win64/Win32
- MinGW-w64 provides a development and runtime environment for 32- and
- 64-bit (x86 and x64) Windows applications using the Windows API and
- the GNU Compiler Collection (gcc).
- .
- This package contains the @@LANGUAGE@@ compiler, supporting
- cross-compiling to 64-bit MinGW-w64 targets, using the Win32
- threading model.
-Build-Profiles: <!stage1>
diff --git a/debian/rules b/debian/rules
index cfba9e6..ecabb89 100755
--- a/debian/rules
+++ b/debian/rules
@@ -25,8 +25,7 @@ export NO_PKG_MANGLE=1
 
 target_version := 10
 target32 := i686-w64-mingw32
-target64 := x86_64-w64-mingw32
-targets := $(target32) $(target64)
+targets := $(target32)
 threads := posix win32
 gnat_arches := alpha amd64 arm64 armel armhf hppa i386 mips64el mipsel ppc64 ppc64el riscv64 s390x sh4 sparc64 x32
 
@@ -67,7 +66,7 @@ ifneq ($(filter stage1,$(DEB_BUILD_PROFILES)),)
     INSTALL_TARGET := install-gcc
 else
 # Build the full GCC.
-    languages := c,c++,fortran,objc,obj-c++
+    languages := c,c++,fortran
 ifneq ($(filter $(DEB_HOST_ARCH),$(gnat_arches)),)
     languages := $(languages),ada
 endif
@@ -246,6 +245,10 @@ CONFFLAGS += \
 # Enable libatomic
 CONFFLAGS += \
 	--enable-libatomic
+# Enable dwarf exceptions
+CONFFLAGS += \
+	--disable-sjlj-exceptions \
+	--with-dwarf2
 # Enable experimental::filesystem and std::filesystem
 CONFFLAGS += \
 	--enable-libstdcxx-filesystem-ts=yes
@@ -314,11 +317,6 @@ ifeq ($(filter stage1,$(DEB_BUILD_PROFILES)),)
 			-B$(build_dir)/$(target32)-$$threads \
 			-D$(upstream_dir) -- \
 			$(CONFFLAGS) --disable-sjlj-exceptions --with-dwarf2; \
-		target=$(target64); \
-		dh_auto_configure \
-			-B$(build_dir)/$(target64)-$$threads \
-			-D$(upstream_dir) -- \
-			$(CONFFLAGS); \
 	done
 else
 	set -e; \
@@ -326,12 +324,7 @@ else
 	dh_auto_configure \
 		-B$(build_dir)/$(target32) \
 		-D$(upstream_dir) -- \
-		$(CONFFLAGS) --disable-sjlj-exceptions --with-dwarf2; \
-	target=$(target64); \
-	dh_auto_configure \
-		-B$(build_dir)/$(target64) \
-		-D$(upstream_dir) -- \
-		$(CONFFLAGS)
+		$(CONFFLAGS) --disable-sjlj-exceptions --with-dwarf2;
 endif
 	touch $@
 
